### PR TITLE
Add golden tests for default and overriden `expandedTitleScale` in `FlexibleSpaceBar`

### PR DESCRIPTION
**Pull request by markusaksli-nc**
Adds golden tests for default and overriden `expandedTitleScale` in `FlexibleSpaceBar` (see https://github.com/flutter/flutter/pull/92160).
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 12 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/93059

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/93059

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 1]
```diff
@@ -542,6 +546,127 @@ void main() {
...
+                    ),
+                    centerTitle: false,
+                    expandedTitleScale: expandedTitleScale,
+                  ),
+                ),
+              ),
+              SliverList(
+                delegate: SliverChildListDelegate(
+                  <Widget>[
+                    for (int i = 0; i < 3; i++)
```
Comment (HansMuller) [inline packages/flutter/test/material/flexible_space_bar_test.dart:635]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer--over-

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 2]
```diff
@@ -542,6 +546,127 @@ void main() {
...
+    );
+
+    // We drag up to fully collapse the space bar.
+    await tester.drag(find.text('Item 0'), const Offset(0, -600.0));
+    await tester.pumpAndSettle();
+
+    final Finder flexibleSpaceBar = find.ancestor(of: find.byType(FlexibleSpaceBar), matching: find.byType(RepaintBoundary).first);
+    // This should match the default behavior
+    await expectLater(
+        flexibleSpaceBar,
```
Comment (HansMuller) [inline packages/flutter/test/material/flexible_space_bar_test.dart:656]: two space indent

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 3]
```diff
@@ -542,6 +546,127 @@ void main() {
...
+        matchesGoldenFile('flexible_space_bar.expanded_title_scale_default.collapsed.png')
+    );
+
+    // We drag down to fully expand the space bar.
+    await tester.drag(find.text('Item 2'), const Offset(0, 600.0));
+    await tester.pumpAndSettle();
+
+    await expectLater(
+        flexibleSpaceBar,
```
Comment (HansMuller) [inline packages/flutter/test/material/flexible_space_bar_test.dart:665]: two space indent

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 4]
```diff
@@ -542,6 +546,127 @@ void main() {
...
```
Comment (HansMuller) [inline packages/flutter/test/material/flexible_space_bar_test.dart:603]: two space indent

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 5]
```diff
@@ -542,6 +546,127 @@ void main() {
...
+      ),
+    );
+
+    // We drag up to fully collapse the space bar.
+    await tester.drag(find.text('Item 0'), const Offset(0, -600.0));
+    await tester.pumpAndSettle();
+
+    final Finder flexibleSpaceBar = find.ancestor(of: find.byType(FlexibleSpaceBar), matching: find.byType(RepaintBoundary).first);
+    await expectLater(
+        flexibleSpaceBar,
```
Comment (HansMuller) [inline packages/flutter/test/material/flexible_space_bar_test.dart:594]: two space indent

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 6]
```diff
@@ -542,6 +546,127 @@ void main() {
...
+                      style: TextStyle(fontSize: titleFontSize,),
+                    ),
+                    centerTitle: false,
+                  ),
+                ),
+              ),
+              SliverList(
+                delegate: SliverChildListDelegate(
+                  <Widget>[
+                    for (int i = 0; i < 3; i++)
```
Comment (HansMuller) [inline packages/flutter/test/material/flexible_space_bar_test.dart:574]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer--over-

**Review by HansMuller — APPROVED** [review]
LGTM This is great, thanks for taking care of it!

Will land this once the trivial formatting stuff is updated.

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 4.
View them at https://flutter-gold.skia.org/cl/github/93059

**Comment by markusaksli-nc** [comment]
Done, thanks for the quick feedback on these!

**Comment by skia-gold** [comment]
Gold has detected about 9 new digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/93059

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
It looks like this change was landed without triaging golden file images: https://flutter-gold.skia.org/search?blame=001000004518&corpus=flutter

**Pull request merged**